src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -228,7 +230,7 @@ public void setNation(Nation nation) throws AlreadyRegisteredException {
 
 	@Override
 	public List<Resident> getResidents() {
-		return Collections.unmodifiableList(residents);
+		return Collections.unmodifiableList(residents.stream().sorted(Comparator.comparingLong(Resident::getJoinedTownAt)).collect(Collectors.toList()));
```

Comment:
I would actually make a separate method that returns a sorted get residents. The general `getResidents()` gets called for a lot of things where it does not need to be sorted, so the sorting is just an unnecessary perf hit.

Reply:
Adressed the performance aspect of it in 27ac19bcb62dd2a4ef3af35941711cedc9035aaa, residents will now only get sorted once at startup, which removes the need for a 2nd method.